text/0004-api-history-schema.md

```diff
@@ -0,0 +1,249 @@
+
+* `position` [Point](structures/point.md)
+
+Set a custom position for the traffic light buttons. Can only be used with `titleBarStyle` set to `hidden`.
+```
+
+After this RFC, they will consult the schema, and also add an API history block:
+
+````markdown
+#### `win.setTrafficLightPosition(position)` _macOS_
+
+<!--
+```YAML history
+added:
+  - pr-url: https://github.com/electron/electron/pull/22533
+```
+-->
+
+* `position` [Point](structures/point.md)
+
+Set a custom position for the traffic light buttons. Can only be used with
+`titleBarStyle` set to `hidden`.
+````
+
+Which, after several PRs, may reach this state:
+
+```yaml
+added:
+  - pr-url: https://github.com/electron/electron/pull/22533
+changes:
+  - pr-url: https://github.com/electron/electron/pull/26789
+    description: Made `trafficLightPosition` option work for `customButtonOnHover` window.
+deprecated:
+  - pr-url: https://github.com/electron/electron/pull/37094
+    breaking-changes-header: deprecated-browserwindowsettrafficlightpositionposition
+```
+
+> [!NOTE]
+>
+> - *"Why no version numbers?"*
+>   - They will be derived from the PRs. This removes the need to change API History
+> on backports.
+>
+> - *"Why no `removed` property?"*
+>   - When an API is removed from Electron, it is also removed from the documentation.
+>
+> For further details, see [**Rationale and alternatives**](#rationale-and-alternatives).
+
+Meanwhile, the Electron website displays this information nicely in a table.
+Users can then use this information to help them upgrade their Electron version,
+migrate from deprecated APIs, etc.
+
+The schema provides information to the developer about what to include in the
+block, the minimum/maximum length of text, the required properties, etc.
+
+Documentation for older APIs will also eventually be changed to include these
+API history blocks.
+
+## Reference-level explanation
+
+The JSON Schema looks like this:
+
+```json
+{
+  "title": "JSON schema for API history blocks in Electron documentation",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$comment": "If you change this schema, remember to edit the TypeScript interfaces in the linting script.",
+  "definitions": {
+    "changeSchema": {
+      "type": "object",
+      "properties": {
+        "pr-url": { "type": "string", "pattern": "^https://github.com/electron/electron/pull/\\d+$" },
+        "breaking-changes-header": { "type": "string", "minLength": 3 },
+        "description": { "type": "string", "minLength": 3, "maxLength": 72 }
+      },
+      "required": [ "pr-url" ],
+      "additionalProperties": false
+    }
+  },
+  "type": "object",
+  "properties": {
+    "added": { "type": "array", "items": { "$ref": "#/definitions/changeSchema" } },
+    "deprecated": { "type": "array", "items": { "$ref": "#/definitions/changeSchema" } },
+    "changes": { "type": "array", "items": { "$ref": "#/definitions/changeSchema" } }
+  },
+  "additionalProperties": false
+}
+```
```

Comment:
Adding `description` fields to the properties might be useful to give a little more description inline (like clarifying that `pr-url` is the URL to the "root" PR to `main`).

Reply:
How's this?

```patch
diff --git a/text/0004-api-history-schema.md b/text/0004-api-history-schema.md
index 28d9075..5f8b3d6 100644
--- a/text/0004-api-history-schema.md
+++ b/text/0004-api-history-schema.md
@@ -111,9 +111,21 @@ The JSON Schema looks like this:
     "changeSchema": {
       "type": "object",
       "properties": {
-        "pr-url": { "type": "string", "pattern": "^https://github.com/electron/electron/pull/\\d+$" },
-        "breaking-changes-header": { "type": "string", "minLength": 3 },
-        "description": { "type": "string", "minLength": 3, "maxLength": 72 }
+        "pr-url": {
+          "description": "URL to the 'root' GitHub Pull Request for the change (i.e. not a backport PR)",
+          "type": "string", "pattern": "^https://github.com/electron/electron/pull/\\d+$",
+          "examples": [ "https://github.com/electron/electron/pull/26789" ]
+        },
+        "breaking-changes-header": {
+          "description": "Heading ID for the change in `electron/docs/breaking-changes.md`",
+          "type": "string", "minLength": 3,
+          "examples": [ "deprecated-browserwindowsettrafficlightpositionposition" ]
+        },
+        "description": {
+          "description": "Short description of the change",
+          "type": "string", "minLength": 3, "maxLength": 72,
+          "examples": [ "Made `trafficLightPosition` option work for `customButtonOnHover`." ]
+        }
       },
       "required": [ "pr-url" ],
       "additionalProperties": false

```

Reply:
Resolved in [`784072b` (#6)](https://github.com/electron/rfcs/pull/6/commits/784072bca8b8ff625eb0f06eb18c3e9301684ff7)